### PR TITLE
total and daily revenue for flash.trade

### PR DESCRIPTION
**Pull request by UjjwalGupta49**
adapter to showcase daily aggregrated revenue and 24 hour aggregated revenue for flash.trade

### CODE REVIEW

**Comment by llamatester** [comment]
The flashtrade.ts adapter exports: 
        
 
 _______________________________________
    
    
    
    ERROR: Use Adapter v2 format for new adapters

File: fees/flashtrade.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    const totalAccured = totalStats.reduce((sum, item) => sum + parseFloat(item.accured), 0);
+    return {
+      timestamp,
+      dailyProtocolRevenue: dailyAccured.toString(),
+      totalProtocolRevenue: totalAccured.toString(),
+    };
+  };
+
+  const adapter: Adapter = {
+    adapter: {
```
Comment (dtmkeng) [inline fees/flashtrade.ts:49]: add `version: 2` above `adapter`

**Review by dtmkeng — CHANGES_REQUESTED** [review]

**Comment by llamatester** [comment]
The flashtrade.ts adapter exports: 
        
 
 _______________________________________
    Fees for 29/5/2024
    _______________________________________
    
    SOLANA 👇
    Backfill start time not defined
    Timestamp: 1716972947 (2024-05-29T08:55:47.000Z)
    Daily protocol revenue: 29.13 k
    Total protocol revenue: 2.57 M

File: fees/flashtrade.ts (hunk 1 above)
Comment (UjjwalGupta49) [reply fees/flashtrade.ts:49]: i've added adapter version 2

**Comment by dtmkeng** [comment]
how many fees ?

**Comment by UjjwalGupta49** [comment]
@dtmkeng what does that mean?

> how many fees ?

**Comment by dtmkeng** [comment]
> @dtmkeng what does that mean?
> 
> > how many fees ?

fees is require field `dailyFees`

**Comment by UjjwalGupta49** [comment]
can you share a reference or a similar example?

**Comment by dtmkeng** [comment]
> can you share a reference or a similar example?

 https://docs.llama.fi/analysts/data-definitions docs

**Comment by llamatester** [comment]
The flashtrade.ts adapter exports: 
        
 
 _______________________________________
    Fees for 4/6/2024
    _______________________________________
    
    TypeError: Cannot read properties of undefined (reading 'reduce')
        at fetchFlashStats (/home/runner/work/dimension-adapters/dimension-adapters/fees/flashtrade.ts:32:37)
        at processTicksAndRejections (node:internal/process/task_queues:95:5)
        at async getChainResult (/home/runner/work/dimension-adapters/dimension-adapters/adapters/utils/runAdapter.ts:31:18)
        at async Promise.all (index 0)
        at async /home/runner/work/dimension-adapters/dimension-adapters/cli/testAdapter.ts:75:21 {
      chain: 'solana'
    }

File: fees/flashtrade.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    return stats
+        .filter(item => item.pool !== "Community.1")
+        .reduce((sum, item) => sum + 0.3 * parseFloat(item.accured), 0);
+}
+
+const fetchFlashStats = async (timestamp: number): Promise<FetchResultFees> => {
+    const totalStatsResponse = await fetchURL(urlTotalStats);
+    const dailyStatsResponse = await fetchURL(urlDailyStats);
+
+    const totalStats: Pool[] = totalStatsResponse.data;
```
Comment (dtmkeng) [inline fees/flashtrade.ts:29]: could you remove `.data` fetchURL is `.data` already

**Review by dtmkeng — CHANGES_REQUESTED** [review]

**Comment by UjjwalGupta49** [comment]
hey @dtmkeng the fees and revenue for flashtrade are different - fees is the 100% of the value and revenue is 30% of fees, the same is being given as input to the adapters for fees and revenue but on the UI is shows fees and revenue as the same. Please have a look

Thanks
image

**Comment by dtmkeng** [comment]
let me check

**Comment by UjjwalGupta49** [comment]
hi there @dtmkeng - there's a recent issue with the volume shown on defilama

it's been showing the same volume for the past three days, please have a look to the issue - thanks.
image
image

**Comment by dtmkeng** [comment]
> hi there @dtmkeng - there's a recent issue with the volume shown on defilama
> 
> it's been showing the same volume for the past three days, please have a look to the issue - thanks. image image

your api it error

[Error: Request failed with status code 403] {
  url: 'https://api.prod.flash.trade/market-stat/revenue-all-time',
  method: 'GET',
  axiosError: 'Forbidden',
  chain: 'solana'
}

**Comment by UjjwalGupta49** [comment]
can you please share you server IP?

**Comment by dtmkeng** [comment]
> can you please share your server IP?

can't we don't have static ip

**Comment by UjjwalGupta49** [comment]
we've updated the endpoint can you please try a re-deployment if that's necessary to update the volume stats.

**Comment by dtmkeng** [comment]
do you have historical data ? will help refil missing  data

**Comment by UjjwalGupta49** [comment]
hey there @dtmkeng on the defillama dashboard for flashtrade th data for fees, volume and revenue looks stale - can you please share the logs to help us better understand the issue.

Thanks you

**Comment by dtmkeng** [comment]
which day is not correct ?

**Comment by UjjwalGupta49** [comment]
it was stale from July 9 th to 13 th for the volume

**Comment by dtmkeng** [comment]
is there historical data will refill ?

**Comment by UjjwalGupta49** [comment]
heres the historiacal data for july 9 to 13 th for deratives volume that was stale -

(values in USD)

2024-07-13 | 18995116.77
2024-07-12 | 41234564.86
2024-07-11 | 102161933.5
2024-07-10 | 103746078
2024-07-09 | 149999449.6

**Comment by UjjwalGupta49** [comment]
also the volume has been stale for the past two days (17, 18) th of july - can you please share the logs for us to improve upon the API.

**Comment by UjjwalGupta49** [comment]
Hey there @dtmkeng the derivatives volume seems to be stagnated from the past few days for flash.trade, we have the correct data to back full stats on our defilama page. Please have look and let me know how can backfill the missing data for volume.

**Comment by UjjwalGupta49** [comment]
Hey there @dtmkeng the derivatives volume seems to be stagnated from the past few days for flash.trade, we have the correct data to back full stats on our defilama page. Please have look and let me know how can backfill the missing data for volume.
Screenshot_20240913-214315

**Pull request merged**